src/ditto/fx/config_gen.py

```diff
@@ -158,17 +159,19 @@ def get_intermediate_size(graph_module: GraphModule) -> int:
         PretrainedConfigGenerationError: If no or multiple intermediate sizes are found.
     """
     values: set[int] = set()
-    values_with_shared_expert: set[int] = set()
+    values_of_shared_experts: set[int] = set()
+    values_of_experts: set[int] = set()
     for node in graph_module.graph.nodes:
         if (linear := Linear.configure_from(node)) and linear.lora_prefix == "mlp_4h_to_h":
-            # TODO: get intermediate size properly for MoE models without shared experts
             if linear.mm.meta.get(EXPERT_TYPE) == "shared_expert":
-                values_with_shared_expert.add(linear.in_features)
+                values_of_shared_experts.add(linear.in_features)
             else:
                 values.add(linear.in_features)
+        if isinstance(node.target, MixtureOfExpertsPlugin):
+            values_of_experts.add(node.target.expert_inter_size * mapping.tp_size)
 
     if len(values) == 0:
-        values = values_with_shared_expert
+        values = values_of_shared_experts if len(values_of_shared_experts) > 0 else values_of_experts
 
     if len(values) == 0:
         raise PretrainedConfigGenerationError("No intermediate size found in graph module")
```

Comment:
It could be more simpler.

```suggestion
    if not (values := values or values_of_shared_experts or values_of_experts):
        raise PretrainedConfigGenerationError("No intermediate size found in graph module")
```